### PR TITLE
Update Snyk Workflow to target other branches (main)

### LINKED ISSUE #20364: Update Snyk Workflow to target other branches
### Description

The current Snyk Workflow focuses specifically on the `main` branches, which is good, but at the same time we are unable to identify important CVEs on previous branches, like for example `21.1.1` which is still on version `2.13.7.Final` of Quarkus.

Unfortunately, the idea of using [Snyk CLI to test multiple branches](https://docs.snyk.io/snyk-cli/test-for-vulnerabilities/grouping-projects-by-branch-or-version), or tags, using `--target-reference` does not work. 
@stianst proposed a [brilliant and elegant solution to the quickstarts](https://github.com/keycloak/keycloak-quickstarts/commit/181cdb9e284a05770d67a9f87c9ce5aaadd327da), which we can benefit from those changes to test more than one branch.

### How?

* Add scheduled.yml file that runs snyk-analysis.yml both for main and latest release branch
* Remove schedule event from snyk-analysis.yml
* Write a script to retrieve the latest release branch

### Proof of Concept (PoC) 

This PoC explores an improvement of the current Snyk Workflow by providing the analysis of CVEs on the previous release branch, overcoming the limitation of using cron jobs on GitHub actions. The current proposal is designed to scan CVEs on `main` and on the previous release branch, currently `release/21.1`.
[At this URL](https://github.com/keycloak-poc/keycloak/actions.), you can see a successful run targeting both branches and below is possible to see the outcome of the changes proposed here:

[Screencast from 2023-05-26 09-14-45.webm](https://github.com/keycloak/keycloak/assets/21758/395ed4ae-70d2-4172-a4fb-529904cd293f)

### PR DESCRIPTION
**Pull request by abstractj**
Depends on:
- https://github.com/keycloak/keycloak/pull/20602
- https://github.com/keycloak/keycloak/pull/20603
- https://github.com/keycloak/keycloak/pull/20604

Closes #20364

### CODE REVIEW

**Comment by abstractj** [comment]
I noticed that the changes proposed will be triggered on pull-request events, that's something that we don't want. I will review and see what can be done.

File: .github/workflows/schedule.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+jobs:
+
+  run-ci:
+    name: Run CI workflow
+    runs-on: ubuntu-latest
+    steps:
+      - run: gh workflow run -R keycloak/keycloak snyk-analysis.yml -r main
```
Comment (stianst) [inline .github/workflows/schedule.yml:14]:
```suggestion
      - name: Run Snyk with main branch
        run: gh workflow run -R keycloak/keycloak snyk-analysis.yml -r main
```

File: .github/workflows/schedule.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        if: github.event_name != 'schedule' || github.repository == 'keycloak/keycloak'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: |
```
Comment (stianst) [inline .github/workflows/schedule.yml:18]:
```suggestion
      - name: Run Snyk with latest release branch
        run: |
```

**Review by stianst — COMMENTED** [review]
LGTM, a smal nitpick would be to add names for the steps to make it more clear what each step is doing

**Comment by abstractj** [comment]
@stianst thank you, updated.

**Review by stianst — APPROVED** [review]

**Pull request merged**